include/deal.II/base/mpi_consensus_algorithms.templates.h

```diff
@@ -692,18 +689,20 @@ namespace Utilities
         // able to test also the non-blocking implementation. This feature
         // is tested by:
         // tests/multigrid/transfer_matrix_free_06.with_mpi=true.with_p4est=true.with_trilinos=true.mpirun=10.output
+
+        const unsigned int n_procs = Utilities::MPI::n_mpi_processes(comm);
```

Comment:
> Attempting to use an MPI routine before initializing MPICH

You can call this function only when MPI is enables. That is why I called `Utilities::MPI::job_supports_mpi()` in the old code.